ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_nat.schema.yml

```diff
@@ -0,0 +1,182 @@
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  ip_nat:
+    documentation_options:
+      filename: data_model/IP NAT
+    type: dict
+    keys:
+      kernel_buffer_size:
+        type: int
+        convert_types:
+          - str
+        min: 1
+        max: 64
+        description: Buffer size in MB
+      pools:
+        type: list
+        primary_key: name
+        convert_types:
+          - dict
```

Comment:
you can remove this as this is in our schema as we are migrating from 3.x to 4.x
essentially this would be used to support

```
ip_nat:
  pools:
    POOL_NAME:
      [stuff]
```

by converting it to

```
ip_nat:
  pools:
    - name:POOL_NAME
       [stuff]
```

in the code but we don't need to support this for new models.

(Keep the conversion for int to str elsewhere though)

Reply:
That was not really clear to me. But understood now.

Reply:
Fixed in [8899c99](https://github.com/aristanetworks/ansible-avd/pull/2747/commits/8899c9909d24d56209fec75f70e38110c176d56b)